lib/phoenix/code_reloader/server.ex

```diff
@@ -13,8 +13,16 @@ defmodule Phoenix.CodeReloader.Server do
     GenServer.call(__MODULE__, :check_symlinks, :infinity)
   end
 
-  def reload!(endpoint) do
-    GenServer.call(__MODULE__, {:reload!, endpoint}, :infinity)
+  def reload!(conn, callback) do
```

Comment:
The reason the ConnRegistry was required is that we are potentially modifying a `conn`, in that we change its state to `:chunked` (using `Plug.Conn. send_chunked`).

So I think merging the registry into Proxy is fine but I think we can't avoid having to track connections, so I believe `Proxy.un/capture` will have to receive/return a connection also (if the user wants to receive messages).